src/aquarium-optimized/Aquarium.cpp

```diff
@@ -205,181 +220,154 @@ bool Aquarium::init(int argc, char **argv)
     }
     toggleBitset.set(static_cast<size_t>(TOGGLE::ENABLEALPHABLENDING));
 
-    int windowWidth  = 0;
-    int windowHeight = 0;
-
-    for (int i = 1; i < argc; ++i)
+    if (result.count("alpha-blending"))
     {
-        std::string cmd(argv[i]);
-        if (cmd == "--num-fish")
+        g.alpha = result["alpha-blending"].as<std::string>();
+        if (g.alpha == "false")
         {
-            mCurFishCount = strtol(argv[i++ + 1], &pNext, 10);
-            if (mCurFishCount < 0)
-            {
-                std::cerr << "Fish count should larger or equal to 0." << std::endl;
-                return false;
-            }
+            toggleBitset.reset(static_cast<size_t>(TOGGLE::ENABLEALPHABLENDING));
         }
-        else if (cmd == "--msaa-sample-count")
+    }
+
+    if (result.count("buffer-mapping-async"))
+    {
+        if (!availableToggleBitset.test(static_cast<size_t>(TOGGLE::BUFFERMAPPINGASYNC)))
         {
-            mContext->setMSAASampleCount(strtol(argv[i++ + 1], &pNext, 10));
+            std::cerr << "Buffer mapping async isn't supported for the backend." << std::endl;
+            return false;
         }
-        else if (cmd == "--enable-instanced-draws")
+
+        toggleBitset.set(static_cast<size_t>(TOGGLE::BUFFERMAPPINGASYNC));
+    }
+
+    if (result.count("disable-control-panel"))
+    {
+        toggleBitset.set(static_cast<size_t>(TOGGLE::DISABLECONTROLPANEL));
+    }
+
+    if (result.count("disable-d3d12-render-pass"))
+    {
+        if (!availableToggleBitset.test(static_cast<size_t>(TOGGLE::DISABLED3D12RENDERPASS)))
         {
-            /*if (!availableToggleBitset.test(static_cast<size_t>(TOGGLE::ENABLEINSTANCEDDRAWS)))
```

Comment:
I hope we can fix instanced rendering, so maybe this option should be preserved.

Reply:
It mentioned to be deprecated, so I removed it. Added this part of code back but I didn't add a new option for this.